### PR TITLE
Huge update to all template engines.

### PR DESCRIPTION
**Pull request by AzureusNation**
Added license to all engines.
Unified all POMs
Created tests for all engines to test the rendering is working.
Updated all dependencies bar JetBrick that is on version 2.1.1 due to a bug.
Updated and created examples for all engines.
Updated and created README's for all engines.
Started to update JavaDocs.

### CODE REVIEW

**Comment by tipsy** [comment]
Thanks, this is great. Will start reviewing tonight.

File: README.md [hunk 1]
```diff
@@ -1,8 +1,21 @@
-spark-template-engines
-======================
+Spark Templating Engines
```
Comment (tipsy) [inline README.md]: Template

File: README.md [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+========================
 
 Repository for different Template engine implementations. 
 
 ## Contributing
 
 Please follow the official Spark [style guidelines](https://github.com/perwendel/spark/tree/master/config) if you want to contribute.
+
+## Current Templating Engines that are supported by Spark
```
Comment (tipsy) [inline README.md]: Template

File: spark-template-freemarker/LICENSE [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright {yyyy} {name of copyright owner}
```
Comment (tipsy) [inline spark-template-freemarker/LICENSE]: {yyyy} {name of copyright owner} -> Make this the same as the one at the main repo?

File: spark-template-handlebars/src/main/java/spark/template/handlebars/HandlebarsTemplateEngine.java [hunk 4]
```diff
@@ -55,24 +38,29 @@ public HandlebarsTemplateEngine() {
...
-                .maximumSize(1000).build();
+        this(new ClassPathTemplateLoader() {{
+            setPrefix(resourceRoot);
+            setSuffix("");
+        }});
+    }
 
-        handlebars = handlebars.with(new GuavaTemplateCache(cache));
+    /**
+     * Constructs the FreeMarkerEngine with a {@link TemplateLoader}
```
Comment (tipsy) [inline spark-template-handlebars/src/main/java/spark/template/handlebars/HandlebarsTemplateEngine.java]: FreeMarkerEngine -> handlebars

File: spark-template-jetbrick/src/main/java/spark/template/jetbrick/JetbrickTemplateEngine.java [hunk 5]
```diff
@@ -26,32 +11,56 @@
 import spark.TemplateEngine;
 
 /**
- * JetTemplateEngine  based on Jetbrick-template
- * See https://github.com/subchen/jetbrick-template-2x
+ * Renders HTML from Route output using FreeMarker.
```
Comment (tipsy) [inline spark-template-jetbrick/src/main/java/spark/template/jetbrick/JetbrickTemplateEngine.java]: Probably not FreeMarker

File: spark-template-jetbrick/src/main/java/spark/template/jetbrick/JetbrickTemplateEngine.java [hunk 6]
```diff
@@ -26,32 +11,56 @@
...
+    /**
+     * Constructs the JetbrickTemplateEngine
+     */
     public JetbrickTemplateEngine() {
-        jetEngine = JetEngine.create();
+        this("/templates");
+    }
+
+    /**
+     * Constructs the FreeMarkerEngine with resource root
```
Comment (tipsy) [inline spark-template-jetbrick/src/main/java/spark/template/jetbrick/JetbrickTemplateEngine.java]: FreeMarker again

**Comment by tipsy** [comment]
Overall, great work! Some notes:
* You should probably do a search/replace for template names in each module, as I'm finding a lot of references to FreeMarker in non-freemarker-modules
* The license should probably be copied from the main repo
* The example we use is not very good at explaining how stuff works. Maybe we should elaborate on it... but I can do this in a separate commit (something like [this](https://gist.github.com/tipsy/276721ab45ee89e08a99407954e2b46c)?

**Comment by AzureusNation** [comment]
Great, I'll get those sorted and if you don't mind with the example. I'm not exactly good at explaining things in words.

**Comment by tipsy** [comment]
The tests seem a bit excessive. Why do need to start a server and test the routing too?
Maybe I'm missing something, but to me it looks like:
```java
String rendered = new FreeMarkerTemplateEngine().render(new ModelAndView(map, "hello.ftl"));
Assert.assertEquals(rendered , expected);
```
Should be enough?

**Comment by tipsy** [comment]
Some of the tests are called `TestRender` , while others are called `RenderTest`.

**Comment by AzureusNation** [comment]
I'll get all of that sorted. I don't think of testing it that way... always overcomplicating things ahah.

**Comment by AzureusNation** [comment]
Hmmm.. Made all the changes and the building passes the tests in both intellij and maven but just not with Travis?

**Comment by tipsy** [comment]
Are you using windows? `\r\n` might be causing issues. 
Try simplifying the test template to something like `The ${templateEngineName} template-engine works!`, you don't really need HTML and newlines to test.

**Comment by AzureusNation** [comment]
ok ill change that and force push again. Do you want me to squash the commits as well?

**Comment by tipsy** [comment]
> Do you want me to squash the commits as well?
That would be nice, yes!

**Comment by AzureusNation** [comment]
Hmm problem, because jade takes the first string of letters as a html node I'm just going to add <h1> to all just so this will work.

**Comment by tipsy** [comment]
@AzureusNation, did you manage to sort it out?

**Comment by AzureusNation** [comment]
Sorry yes it's all done. Just been busy at work.

**Comment by tipsy** [comment]
Fixing this now

**Pull request closed without merging**